File: unit-2/sprint-2/day-3-RecyclerView-II/assignments/BirdTest/app/src/main/java/com/example/animal/MainActivity.java
package com.example.animal;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.widget.Toast;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity implements  ItemClickListener{

    private RecyclerView mRecyclerView;
    private ArrayList<Model> modelList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mRecyclerView = findViewById(R.id.rvRecyclerView);
        buildData();
        setRecyclerAdapter();
    }

    private void buildData() {
        modelList = new ArrayList<>();
        for (int i = 0; i < 100; i++) {
            if (i % 2 == 0)
                modelList.add(new Model(R.drawable.crow,"Crow"));
            if (i % 2 == 1)
                modelList.add(new Model(R.drawable.owl,"Owl"));
            if (i % 2 == 0)
                modelList.add(new Model(R.drawable.peacock,"Peacock"));
            if (i % 2 == 1)
                modelList.add(new Model(R.drawable.piegon,"Piegon"));

        }
    }
    private void setRecyclerAdapter(){
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        mRecyclerView.setLayoutManager(layoutManager);
        ModelAdaptor modelAdaptor = new ModelAdaptor(modelList,this);
        mRecyclerView.setAdapter(modelAdaptor);
    }

    @Override
    public void onItemClicked(Model model) {
        Toast .makeText(this,model.getName(),Toast.LENGTH_SHORT).show();
    }
}

File: unit-5/sprint-2(Coding)/Project-1/app/src/main/java/com/masai/sprint_2/ui/MainActivity.kt
package com.masai.sprint_2.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.masai.sprint_2.R
import com.masai.sprint_2.ui.adapter.MainAdapter
import com.masai.sprint_2.ui.viewmodel.MainViewModel
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers.Main
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    private lateinit var mainViewModel:MainViewModel
    private lateinit var mainAdapter: MainAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        mainViewModel = ViewModelProvider(this).get(MainViewModel::class.java)
        setAdapter()
        mainViewModel.searchPeopleList().observe(
            this,{
                it?.let {
                    CoroutineScope(Main).launch {
                        mainAdapter.submitData(it)
                    }
                }
            }
        )
    }

    private fun setAdapter() {
        mainAdapter = MainAdapter()
        val linearLayoutManager = LinearLayoutManager(this)
        recyclerView.apply {
            layoutManager = linearLayoutManager
            this.adapter = mainAdapter
        }
    }
}